src/Lookup.hs

```diff
@@ -28,6 +29,27 @@ lookupInEnv path@(SymPath (p : ps) name) env =
         Just parent -> lookupInEnv path parent
         Nothing -> Nothing
 
+lookupInterfaceOrType :: Context -> SymPath -> Maybe Binder
+lookupInterfaceOrType ctx path =
+  let typeEnv = (getTypeEnv (contextTypeEnv ctx))
+   in lookupBinder path typeEnv
+
+lookupGlobalBinding :: Context -> SymPath -> Maybe Binder
+lookupGlobalBinding ctx path =
+  let global = (contextGlobalEnv ctx)
+   in lookupBinder path global
+
+lookupContextualBinding :: Context -> SymPath -> Maybe Binder
+lookupContextualBinding ctx path =
+  let ctxEnv = contextEnv ctx
+   in lookupBinder path ctxEnv
+
+lookupEverywhere :: Context -> SymPath -> Maybe [Binder]
```

Comment:
I think this one should still be called something with "multiLookup" since it returns many Binders.

Reply:
4d30c24